scRNA-seq/04-dimension_reduction_scRNA.Rmd

```diff
@@ -31,225 +44,255 @@ library(magrittr)
+The data was generated with the 10Xv3.1 chemistry, and processed with Cell Ranger and 10X Genomics standard pipeline.
+https://support.10xgenomics.com/single-cell-gene-expression/datasets/4.0.0/Parent_NGSC3_DI_HodgkinsLymphoma
+
+There are a variety of files that you will often see as part of the standard output from Cell Ranger, which are described in detail in [10X Genomics documentation](https://support.10xgenomics.com/single-cell-gene-expression/software/pipelines/latest/output/overview).
+We have included some of these in the `data/hodkins/cellranger` directory, including the `web_summary.html` file that includes some similar QC statistics to those we generated with `alevinQC`.
+The main file we will be working with are the feature by barcode matrices.
+Cell Ranger does some filtering on its own, but we will start with the raw data.
+
+
+
 ```{r filepaths}
 # main data directory
-data_dir <- file.path("data", "tabula-muris")
+data_dir <- file.path("data", "hodgkins")
 
-# Path to the single-sample alevin results
-alevin_file <- file.path(data_dir, "alevin-quant", 
-                         "10X_P4_3", "alevin", "quants_mat.gz")
+# Path to the Cell Ranger matrix
+raw_matrix_dir <- file.path(data_dir, "cellranger", 
+                            "raw_feature_bc_matrix")
 
-# normalized multi-sample Tabula Muris Data
-tm_normalized_file <- file.path(data_dir, "normalized",
-                                "TM_normalized.rds" )
+# Path to mitochondrial genes table
+mito_file <- file.path(data_dir, "hs_mitochondrial_genes.tsv")
 
-# Path to the metadata file 
-metadata_file <- file.path(data_dir, 
-                           "TM_droplet_metadata.csv")
+# Directory and file to save output
+normalized_dir <- file.path(data_dir, "normalized")
+if (!dir.exists(normalized_dir)) {
+  dir.create(normalized_dir, recursive = TRUE)
+}
+
+output_sce_file <- file.path(normalized_dir, "normalized_hodgkins_sce.rds")
 
-# Mitochondrial gene table
-mito_file <- file.path(data_dir,
-                       "mm_mitochondrial_genes.tsv")
 ```
 
-## Single sample QC and normalization
+## Reading Cell Ranger data
+
+Cell Ranger output includes count data in two main formats.
+The first is a folder with a feature list, a barcode list, and a sparse matrix in ["Matrix Exchange" format](https://math.nist.gov/MatrixMarket/formats.html). 
+The `DropletUtils::read10xCounts()` function takes this directory and reads in the data from these three files, assembling the `SingleCellExperiment` object we have worked with before. 
+
+Alternatively, we could use the `HDF5` format file that Cell Ranger outputs as a file with the `.h5` extension, which contains the same data. 
+For whatever reason, the way you read the data affects how it is stored in R, with the net result that reading from the directory results in smaller objects in R, so that is what we will do here. 
```

Comment:
```suggestion
For whatever reason, the way you read the data affects how it is stored in R. 
Reading from the directory results in smaller objects in R, so that is what we will do here. 
```